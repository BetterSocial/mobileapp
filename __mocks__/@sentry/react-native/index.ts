export const addBreadcrumb = jest.fn();
