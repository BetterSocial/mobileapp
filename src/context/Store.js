import * as React from 'react';

import {channelReducer, channelState} from './reducers/channelReducer';
import {clientReducer, clientState} from './reducers/clientReducer';
import {discoveryReducer, discoveryState} from './reducers/discoveryReducer';
import {domainReducer, domainState} from './reducers/domainReducer';
import {feedsReducer, feedsState} from './reducers/FeedReducer';
import {followingReducer, followingState} from './reducers/followingReducer';
import {generalComponentReducer, generalComponentState} from './reducers/generalComponentReducer';
import {groupChatReducer, groupChatState} from './reducers/groupChat';
import {initialStartupReducer, initialStartupState} from './reducers/intialStartupReducer';
import {localCommunityReducer, localCommunityState} from './reducers/localComunityReducer';
import {myProfileFeedReducer, myProfileFeedState} from './reducers/myProfileFeedReducer';
import {myProfileReducer, myProfileState} from './reducers/myProfileReducer';
import {newsReducer, newsState} from './reducers/newsReducer';
import {otherProfileFeedReducer, otherProfileFeedState} from './reducers/OtherProfileFeedReducer';
import {topicsReducer, topicsState} from './reducers/topicsReducer';
import {unReadMessageReducer, unReadMessageState} from './reducers/unReadMessageReducer';
import {usersReducer, usersState} from './reducers/userReducer';
import {commentReducer, commentState} from './reducers/commentReducer';

/**
 *
 * @typedef {Object} MainContext
 * @property {import('./reducers/discoveryReducer').DiscoveryState} discovery
 */
const Store = ({children}) => {
  const rootReducer = {
    channel: React.useReducer(channelReducer, channelState),
    client: React.useReducer(clientReducer, clientState),
    discovery: React.useReducer(discoveryReducer, discoveryState),
    feeds: React.useReducer(feedsReducer, feedsState),
    following: React.useReducer(followingReducer, followingState),
    generalComponent: React.useReducer(generalComponentReducer, generalComponentState),
    groupChat: React.useReducer(groupChatReducer, groupChatState),
    localCommunity: React.useReducer(localCommunityReducer, localCommunityState),
    myProfileFeed: React.useReducer(myProfileFeedReducer, myProfileFeedState),
    otherProfileFeed: React.useReducer(otherProfileFeedReducer, otherProfileFeedState),
    news: React.useReducer(newsReducer, newsState),
    profile: React.useReducer(myProfileReducer, myProfileState),
    topics: React.useReducer(topicsReducer, topicsState),
    unReadMessage: React.useReducer(unReadMessageReducer, unReadMessageState),
    users: React.useReducer(usersReducer, usersState),
    domains: React.useReducer(domainReducer, domainState),
    initialStartup: React.useReducer(initialStartupReducer, initialStartupState),
    comments: React.useReducer(commentReducer, commentState)
  };
  return <Context.Provider value={rootReducer}>{children}</Context.Provider>;
};
export const Context = React.createContext(null);
export const useAppContext = () => React.useContext(Context);

export default Store;
