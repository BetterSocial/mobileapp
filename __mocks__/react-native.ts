import * as RN from 'react-native';

RN.Animated.timing = () => ({
    start: () => jest.fn()
})

module.exports = RN