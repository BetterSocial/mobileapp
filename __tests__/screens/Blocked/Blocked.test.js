import React from 'react'
import {render} from '@testing-library/react-native'
import BlockedTab from "../../../src/screens/Blocked"
import Store from '../../../src/context/Store'

jest.mock('@react-navigation/native', () => ({
    createNavigatorFactory: jest.fn()
}))

jest.mock('@react-navigation/material-top-tabs', () => ({
    createMaterialTopTabNavigator: jest.fn().mockImplementation(() => ({Navigator: jest.fn()}))
}))

jest.mock('../../../src/hooks/useAfterInteractions', () => ({
    useAfterInteractions: () => ({
        transitionRef: {current: {animateNextTransition: jest.fn()}}, interactionsComplete: true
    })
}))

describe('Blockeg page should run correctly', () => {
    it('Blocked tab hhould match snapshot', () => {
        const mockNavigation = {
            addListener: jest.fn()
        }
        const {toJSON} = render(<BlockedTab navigation={mockNavigation} />, {wrapper: Store})
        expect(toJSON).toMatchSnapshot()
    })
})