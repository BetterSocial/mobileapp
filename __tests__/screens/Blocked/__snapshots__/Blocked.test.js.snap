// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blockeg page should run correctly Blocked tab hhould match snapshot 1`] = `[Function]`;
